docs/coding-standards/coding-style.md

```diff
@@ -0,0 +1,160 @@
+},
+```
+
+What about indenting long method names?
+```
+doSomething (
+  param1: string,
+  param2: number, // can add a comment here
+  param3 = false
+): void {
+  // do something
+}
+```
+
+### Component IDs
+
+IDs should be named in kebab-case.
+For example:
+```
+<MyComponent id="my-component" />
+```
+
+### Class Names
+Classes should be named "mostly" in kebab-case.
+For example:
+```
+<div class="my-class">
+</div>
+```
+
+Exception: some "deep" elements may use double-underscores.
+For example:
+```
+<div class="v-input v-text-field">
+  <div class="v-input__control">
+    <div class="v-input__slot">
+      <div class="v-text-field__slot">
+      </div>
+    </div>
+    <div class="v-text-field__details">
+      <div class="v-messages">
+        <div class="v-messages__wrapper">
+          <div class="v-messages__message">
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</div>
+```
+
+### Variables/Properties
+
+Variables/properties should be named in camelCase.
+Why: It's the standard for most languages and it's the default in most IDEs.
+For example:
+```
+const myVariable = 'my value'
+```
+
+Enums, interfaces, types (and some constants) should be named in PascalCase.
+Why: It's the standard for most languages and it's the default in most IDEs.
+For example:
+```
+export const EmptyOfficer: OfficerIF = {
+  firstName: '',
+  lastName: '',
+  middleInitial: ''
+}
+```
+
+### Functions (Methods)
+
+_FUTURE: add something here_
+
+### Components
+
+Components in the template should use PascalCase. Why: It helps differentiate custom Vue components from native Vuetify or HTML components in the template. Also, some IDEs allow you to F12 on the component name to jump to the component definition.
+
+For example:
+```
+<MyComponent />
+```
+
+### Code Comments
+
+Where it makes sense, use JSDoc to say what the function does / is for, as well as to document the formal parameters. For example:
+```
+/**
+  * Removes the specified properties from nested objects.
+  * @param baseObj the base object
+  * @param keys the nested object keys which to omit properties from
+  * @param prop the properties to be removed
+  */
+omitProps (baseObj: any, keys: Array<string>, prop: Array<string>): any {
+  const parsedObj: any = {}
+  Object.keys(baseObj).forEach(key => {
+    parsedObj[key] = omit(baseObj[key], prop)
+  })
+  return parsedObj
+}
+```
+
+### Whitespace
+
+Where it makes sense (eg, to separate logical blocks of code), use whitespace to make the code easier to read. You should also use whitespace between CSS blocks.
+
+There should be whitespace between the template, script and style blocks in a Vue file, but no extra blank lines within the blocks.
+For example:
+```
+<template>
+</template>
+
+<script lang="ts">
+</script>
+
+<style lang="scss">
+</style>
+```
+
+### Consistency
+
+Being consistent with surrounding code _may_ override the above guidelines!
+
+## Back End (Python)
+
+_add some examples/guidelines here_
```

Comment:
This is excellent @severinbeauvais thanks for putting this together.  My only suggestion, and it's very minor, is to separate the Python and JavaScript style guides into separate files.   I know for Python there are a number of great guides that have already been developed.  What do you think of adopting a well-respected style guide rather than developing our own?  At the very least it would save us (you!) some time.

Reply:
My intention was to document our current coding style -- for developers new to our code, and to pre-empt some basic PR review comments.

Ultimately, it would be great to use advanced linter options, or a code prettifier, to do a lot of this for us (or to police our work).

Reply:
And we're part way there! The latest eslint package that we're importing (and presets) takes care of managing a lot of code style issues (including formatting).